modules/tinymce/src/plugins/table/main/ts/core/TableSections.ts

```diff
@@ -80,11 +81,27 @@ const switchRowSection = (dom: DOMUtils, rowElm: HTMLElement, newSectionName: st
   }
 };
 
-const switchCellType = (dom: DOMUtils, cells: ArrayLike<HTMLTableCellElement>, newCellType: string, scope: 'col' | null) =>
-  Arr.each(cells, (c) => {
-    const newCell = Util.getNodeName(c) !== newCellType ? dom.rename(c, newCellType) : c;
+const renameCell = (editor: Editor, cell: HTMLTableCellElement, newCellType?: string): HTMLTableCellElement => {
+  if (Type.isNonNullable(newCellType) && Util.getNodeName(cell) !== newCellType) {
+    const newCellElm = editor.dom.rename(cell, newCellType) as HTMLTableCellElement;
+    Events.fireNewCell(editor, newCellElm);
+    return newCellElm;
+  } else {
+    return cell;
+  }
+};
+
+const switchCellType = (editor: Editor, cell: HTMLTableCellElement, newCellType?: string, scope?: 'col' | null): HTMLTableCellElement => {
+  const dom = editor.dom;
+  const newCell = renameCell(editor, cell, newCellType);
+  if (!Type.isUndefined(scope)) {
     dom.setAttrib(newCell, 'scope', scope); // mutates
-  });
+  }
+  return newCell;
+};
+
+const switchCellsType = (editor: Editor, cells: ArrayLike<HTMLTableCellElement>, newCellType?: string, scope?: 'col' | null) =>
```

Comment:
Can we make that scope arg either not optional or not nullable? At the moment there are two separate "don't want to set a scope" cases and unless there's a big difference between the two that I'm not seeing I feel like we should consolidate that

Reply:
There is a big difference between the two:
- null removes the  scope attribute
- undefined doesn't change the value